src/coredump/coredumpctl.c

```diff
@@ -197,6 +201,22 @@ static int parse_argv(int argc, char *argv[]) {
 
                         break;
 
+                case 'S':
+                        r = parse_timestamp(optarg, &arg_since);
+                        if (r < 0) {
+                                log_error("Failed to parse timestamp: %s", optarg);
+                                return -EINVAL;
+                        }
```

Comment:
Pushed updated commit with this suggested change. I guess journalctl could use these improvements as well because at the moment -EINVAL is returned in these identical cases.